src/Fantomas/CodePrinter.fs

```diff
@@ -1339,7 +1339,7 @@ and genExpr astContext synExpr ctx =
 
             fun ctx ->
                 if List.exists isIfThenElseWithYieldReturn xs
-                   || List.forall isSynExprLambda xs then
+                   || List.forall (fun x -> isSynExprLambda x || isIfThenElse x) xs then
```

Comment:
As `isSynExprLambda` is only used here, I would change that function to also take `SynExpr.IfThenElse` and rename it to something like `isDangerousExprInSingleLineArrayOrList` (or some better name 😅)

Reply:
Ah yes, I thought about that, too. But somehow expected that separate functions would be preferred.
Anyway, naming is hard...

Reply:
Thanks!